api/modules/Catalog/Catalog.Application/Products/EventHandlers/ProductReadEventHandler.cs

```diff
@@ -0,0 +1,17 @@
+using FSH.WebApi.Catalog.Domain.Events;
+using Microsoft.Extensions.Logging;
+using MediatR;
+
+namespace FSH.WebApi.Catalog.Application.Products.EventHandlers;
+
+public class ProductReadEventHandler(ILogger<ProductReadEventHandler> logger) : INotificationHandler<ProductRead>
```

Comment:
Usually only Commands should have Events and it's handlers. I think there is no need to fire a domain event when there is read query coming in. I don't see the use case of it.

Reply:
done